### DISCUSSION TITLE
Q&A Engineering #31: How does learning rate warmup protect transformer attention layers in early training?

### DISCUSSION THREAD

**Discussion by BerattCelikk** [BerattCelikk/daily-engineering-log#393] [Q&A] [answered]
How does learning rate warmup protect transformer attention layers in early training?

**Comment by BerattCelikk** [comment, marked as answer]
At initialization, random weights produce erratic attention logits. A low initial learning rate prevents the optimizer from taking catastrophically large steps before moving variance estimates stabilize.
